include/vcpkg/metrics.h

```diff
@@ -22,18 +84,23 @@ namespace vcpkg
 
         void track_metric(const std::string& name, double value);
         void track_buildtime(const std::string& name, double value);
-        void track_property(const std::string& name, const std::string& value);
-        void track_property(const std::string& name, bool value);
+
+        void track_property(DefineMetric metric);
```

Comment:
Note that in almost all current callers, they do something like `track_property(metric, "a literal")` which calls the `bool` overload, not the string overload. If you are messing with this I think you want to fix that?

That is, I don't see any reason that these need to be overloads. Why not `track_define`, `track_string`, and `track_bool`?